### PR TITLE
Add renew authentication endpoint

### PR DESCRIPTION
**Pull request by lbalmaceda**
Renew the authentication using a `refresh_token`.
Closes https://github.com/auth0/auth0-java/issues/48

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -209,6 +209,24 @@ try {
 }
 ```
 
+### Renew Authentication
+
+Creates a new request to renew the authentication and get fresh new credentials using a valid Refresh Token.
```
Comment (nikolaseu) [inline README.md]: can I create something "old" ? xD just kidding, but the term "(re)new" is repeated several times, maybe we can remove the first

Comment (nikolaseu) [inline README.md]: do we need to specify like in `### Request Token for Audience - /oauth/token` that this is the "new" /oauth/token refresh token?

File: src/main/java/com/auth0/client/auth/AuthAPI.java [hunk 2]
```diff
@@ -35,6 +36,7 @@
     private final String clientId;
     private final String clientSecret;
     private final String baseUrl;
```
Comment (nikolaseu) [inline src/main/java/com/auth0/client/auth/AuthAPI.java]: do we still need this one?

File: src/main/java/com/auth0/client/auth/AuthAPI.java [hunk 3]
```diff
@@ -54,6 +56,7 @@ public AuthAPI(String domain, String clientId, String clientSecret) {
         if (baseUrl == null) {
             throw new IllegalArgumentException("The domain had an invalid format and couldn't be parsed as an URL.");
         }
+        this.httpUrl = HttpUrl.parse(baseUrl);
```
Comment (nikolaseu) [inline src/main/java/com/auth0/client/auth/AuthAPI.java]: `createBaseUrl` already uses `HttpUrl.parse`, so maybe make that return that httpUrl directly?
then `baseUrl` (if really required) can be obtained from `baseUrl = httpUrl.toString()`

File: src/main/java/com/auth0/client/auth/AuthAPI.java (hunk 2 above)
Comment (lbalmaceda) [inline src/main/java/com/auth0/client/auth/AuthAPI.java]: Not really, was used by the `AuthorizeUrlBuilder` but now that I see we can replace that parameter with the `HttpUrl`. Let me fix it. 👍

**Comment by nikolaseu** [comment]
Looks good

**Pull request merged**